design/KafkaDesign.md

```diff
@@ -0,0 +1,27 @@
+# Kafka Documentation
+
+Kruize has a BulkService feature which designed to streamline the detection, management, and optimization of containerized environments. It relies on the REST APIs without a message broker, which introduces latency when handling multiple parallel requests.
+Kafka replaces this existing REST API service for faster, asynchronous communication and to ensure seamless handling of recommendations at scale.
+
+Kruize Kafka Producer internally uses the BulkService and publishes recommendations in the form of message in three topics namely, `recommendations-topic`, `error-topic` and `summary-topic`.
+
+## Kafka Flow
+
+1. Currently, Kruize Kafka Module works as a Producer only i.e. to invoke the Kafka Service user needs to hit a REST API POST request with the same input as the one for the BulkService.
+2. On receiving the request, BulkService will return the `job_id` back and in the background starts the following tasks:
+    - First, does a handshake with the datasource.
+    - Using queries, it fetches the list of namespaces, workloads, containers of the connected datasource.
+    - Creates experiments, one for each container *alpha release.
+    - Triggers `generateRecommendations` for each container.
+    - Once the above step returns a success response, Kafka Producer is invoked and the recommendations are pushed into the `recommendations-topic`
```

Comment:
@khansaad Can you please add examples of the messages that are push'd to the topics

Reply:
Done